### PR TITLE
[SYCL][ESIMD] dgetrfnp regression test

### PR DESCRIPTION
**Pull request by michael-chuvelev**
The test creates ESIMD program implementing LU decomposition without pivoting for certain small sizes, having a number of code branches and extensive inlining.

### CODE REVIEW

File: SYCL/ESIMD/regression/dgetrf.cpp [hunk 1]
```diff
@@ -0,0 +1,459 @@
...
+    int64_t *ipiv_gpu;
+    int64_t *info_gpu;
+    CHECK(a_gpu = static_cast<double *>(malloc_shared(stride_a * batch * sizeof(double), device, context)), !a_gpu);
+    CHECK(ipiv_gpu = static_cast<int64_t *>(malloc_shared(stride_ipiv * batch * sizeof(int64_t), device, context)), !ipiv_gpu);
+    CHECK(info_gpu = static_cast<int64_t *>(malloc_shared(batch * sizeof(int64_t), device, context)), !info_gpu);
+
+    memcpy(a_gpu, a, stride_a * batch * sizeof(double));
+
+    sycl::nd_range<1> range(sycl::range<1>{static_cast<size_t>(batch)}, sycl::range<1>{1});
+    auto event = queue.submit([&](handler &cgh) {
```
Comment (DenisBakhvalov) [inline SYCL/ESIMD/regression/dgetrf.cpp:324]: Please wrap `submit` call in `try/catch` block to handle `cl::sycl::exception` similar to other ESIMD tests.

**Review by DenisBakhvalov — CHANGES_REQUESTED** [review]
In general, LGTM.
I didn't dig too much into the algorithm itself but it would be nice if you could add a short description of what this test is checking in the header of the file.

File: SYCL/ESIMD/regression/dgetrf.cpp (hunk 1 above)
Comment (michael-chuvelev) [reply SYCL/ESIMD/regression/dgetrf.cpp:324]: Hi Denis,

I've added short description and wrapped submit.

**Comment by michael-chuvelev** [comment]
@kbobrovs - LLVM copyright has been added

File: SYCL/ESIMD/regression/dgetrf.cpp [hunk 2]
```diff
@@ -0,0 +1,479 @@
...
+      cgh.parallel_for<class dgetrfnp_batch_strided>(
+          range, [=](nd_item<1> id) SYCL_ESIMD_KERNEL {
+            int i = id.get_global_id(0);
+            dgetrfnp_esimd(m, n, &a_gpu[i * stride_a], lda,
+                           &ipiv_gpu[i * stride_ipiv], &info_gpu[i]);
+          });
+    });
+    event.wait();
+  } catch (const sycl::exception &e) {
+    std::cout << "*** EXCEPTION caught: " << e.what() << "\n";
```
Comment (DenisBakhvalov) [inline SYCL/ESIMD/regression/dgetrf.cpp:334]: I think there is a memory leak here. Calls to `free` should be added.

Comment (michael-chuvelev) [reply SYCL/ESIMD/regression/dgetrf.cpp:334]: Done.

**Review by DenisBakhvalov — APPROVED** [review]

**Review by kbobrovs — APPROVED** [review]
Thanks for updating the copyright

**Pull request merged**